websockets.md

```diff
@@ -0,0 +1,101 @@
+# WebSockets improvements
+
+## Upgrade Response Details
+
+Prior to .NET 7, WebSocket's HTTP upgrade response details were hidden inside `ClientWebSocket` implementation, and all connect errors would surface as `WebSocketException` without much details beside the exception message. However, the information about response headers and status code might be important in both failure and success scenarios.
+
+In case of failure, HTTP status code can help to distinguish between retriable and non-retriable errors (e.g. server doesn't support WebSockets at all, or it was just a small transient error). Headers might also contain additional information on how to handle the situation. The headers are also useful even in case of a success WebSocket connect, e.g. they can contain a token tied to a session, or some info related to subprotocol version, or that the server can go down soon, etc.
```

Comment:
```suggestion
In case of failure, HTTP status code can help to distinguish between retriable and non-retriable errors (e.g. server doesn't support WebSockets at all, or it was just a transient error). Headers might also contain additional information on how to handle the situation. The headers are also useful even in case of a successful WebSocket connect, e.g. they can contain token tied to a session, information related to subprotocol version, or that the server can go down soon.
```